src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -193,20 +192,7 @@ internal static bool IsLegalCSharp73SpanStackAllocPosition(this SyntaxNode node)
         }
 
         internal static CSharpSyntaxNode AnonymousFunctionBody(this SyntaxNode lambda)
-        {
-            switch (lambda.Kind())
-            {
-                case SyntaxKind.SimpleLambdaExpression:
-                    return ((SimpleLambdaExpressionSyntax)lambda).Body;
-                case SyntaxKind.ParenthesizedLambdaExpression:
-                    return ((ParenthesizedLambdaExpressionSyntax)lambda).Body;
-                case SyntaxKind.AnonymousMethodExpression:
-                    return ((AnonymousMethodExpressionSyntax)lambda).Block;
-
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(lambda.Kind());
-            }
-        }
+            => ((AnonymousFunctionExpressionSyntax)lambda).Body;
```

Comment:
we used to throw before if we didn't have an anonymous function.  Now we'll still just throw.  It will jsut be a class-cast exception instead of an unexpected-value exception.  I think that's fine for an internal method.